geopy/geocoders/geolake.py

```diff
@@ -0,0 +1,163 @@
+from geopy.compat import urlencode
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.util import logger
+
+__all__ = ("Geolake", )
+
+
+class Geolake(Geocoder):
+    """Geocoder using the Geolake API.
+
+    Documentation at:
+        https://geolake.com/docs/api
+
+    Terms of Service at:
+        https://geolake.com/terms-of-use
+
+    .. versionadded:: 1.18.0
+    """
+
+    structured_query_params = {
+        'country',
+        'state',
+        'city',
+        'zipcode',
+        'street',
+        'address',
+        'houseNumber',
+        'subNumber',
+    }
+
+    api_path = '/v1/geocode'
+
+    def __init__(
+            self,
+            api_key,
+            domain='api.geolake.com',
+            scheme=None,
+            timeout=DEFAULT_SENTINEL,
+            proxies=DEFAULT_SENTINEL,
+            user_agent=None,
+            format_string=None,
+            ssl_context=DEFAULT_SENTINEL,
+    ):
+        """
+
+        :param str format_string:
+            See :attr:`geopy.geocoders.options.default_format_string`.
+
+        :param str api_key: The API key required by Geolake
+            to perform geocoding requests. You can get your key here:
+            https://geolake.com/
+
+        :param str domain: Currently it is ``'api.geolake.com'``, can
+            be changed for testing purposes.
+
+        :param str scheme:
+            See :attr:`geopy.geocoders.options.default_scheme`.
+
+        :param int timeout:
+            See :attr:`geopy.geocoders.options.default_timeout`.
+
+        :param dict proxies:
+            See :attr:`geopy.geocoders.options.default_proxies`.
+
+        :param str user_agent:
+            See :attr:`geopy.geocoders.options.default_user_agent`.
+
+        :type ssl_context: :class:`ssl.SSLContext`
+        :param ssl_context:
+            See :attr:`geopy.geocoders.options.default_ssl_context`.
+
+        """
+        super(Geolake, self).__init__(
+            format_string=format_string,
+            scheme=scheme,
+            timeout=timeout,
+            proxies=proxies,
+            user_agent=user_agent,
+            ssl_context=ssl_context,
+        )
+
+        self.api_key = api_key
+        self.domain = domain.strip('/')
+        self.api = '%s://%s%s' % (self.scheme, self.domain, self.api_path)
+
+    def geocode(
+            self,
+            query,
+            country_codes=None,
```

Comment:
Could you please add a test case which uses this parameter?

Reply:
Definitely, should be included in the latest commit.